packages/viewer/src/components/viewer/viewer.tsx

```diff
@@ -369,6 +366,41 @@ export class Viewer {
     });
   }
 
+  private connectStream(connection: Promise<Disposable>): Promise<void> {
+    return new Promise(async resolve => {
+      try {
+        const dispose = await connection;
+        this.streamDisposables.push(dispose);
+        resolve();
+      } catch (e) {
+        this.errorMessage = 'Unable to maintain connection to Vertex';
+        throw new WebsocketConnectionError(this.errorMessage, e);
+      }
+    });
+  }
+
+  private reconnectStreamingClient(streamId: UUID.UUID): Promise<string> {
+    this.streamDisposables.forEach(d => d.dispose());
+
+    this.stream = new StreamApi(
+      new WebSocketClient(),
+      this.stream.getUrlProvider()
+    );
```

Comment:
Looking again at this, can't we just reuse the existing `StreamApi` and `stream.connect` for reconnections? Would make the implementation quite a bit simpler because we're just leveraging existing code. We'd also not need to pass down those `StreamProvider`s.

The implementation of `reconnectStreamingClient` would look something like:

```ts
  private reconnectStreamingClient(streamId: UUID.UUID): Promise<string> {
    return new Promise(async resolve => {
      await this.connectStream(
        this.commands.execute('stream.connect', {
          // You can pass `loadedSceneId` here instead of having a new `stream.reconnectWebSocket` command
          sceneId: this.loadedSceneId,
        })
      );

      // We can just use `StreamApi.reconnect` without having to define a new command.
      this.stream.reconnect({
        streamId: { hex: streamId },
        dimensions: this.dimensions,
      });
      resolve(streamId);
    });
  }
```

You don't need an array of stream disposables or the call to `setupStreamListeners` anymore.